cpp/ps0008/main (13).cpp

```diff
@@ -8,58 +7,85 @@ int Decode (string msg , int key);
 
 int main()
 {
+    // prompting user for valid ciphertext and key
+    string message;
+    int key;
     
-string message ;
-int key;
-
-cout<<"enter your message \n";
-getline(cin , message);
-
-
-cout<<"enter your key \n";
-cin >>key;
-
-cout<<"message = "<<Decode(message , key)<<"\n";
+    cout<<"enter your message \n";
+    getline(cin , message); // taking message 1/1
+    
+    // Checks if key is valid 0.5/3
+    string tmp_key;
+    bool validkey = false;
+    int key_len;
+    while (!validkey) {
+        
+        cout << "key:";
+        getline(cin , tmp_key);
+        key_len = tmp_key.length();
+        
+        if (key_len==0 || (key_len==1 && (tmp_key[0]=='+' || tmp_key[0]=='-')))
+            continue;
+            
+        validkey = true;
+        int i = 0;
+        for (char c : tmp_key) {
+            if (i==0 && (c=='-' || c=='+'))
+            {
+                ++i;
+                continue;
+            }
+            if (!isdigit(c)) {
+                validkey = false;
+                break;
+            }
+            ++i;
+        }
+    }
+    key = stoi(tmp_key);
+    
+    // invoke your decrypt function to return the plaintext 1/1
+    Decode(message , key);
```

Comment:
- [x] invoke a function for decryption process
- [x] printed the plaintext (actual message before encryption)